basic/src/test/java/org/jboss/hal/testsuite/test/configuration/undertow/UndertowOperations.java

```diff
@@ -139,153 +73,115 @@ public void removeSecurityRealm(String realmName) throws CommandFailedException,
         administration.reloadIfRequired();
     }
 
-    public String createHTTPServerHost(String httpServer) throws InterruptedException, IOException, TimeoutException {
-        String hostName = RandomStringUtils.randomNumeric(6);
-        createHTTPServerHost(httpServer, hostName);
-        return hostName;
-    }
-
-    public String createHTTPServerHost(String httpServer, String hostName) throws InterruptedException, TimeoutException, IOException {
-        ResourceAddress address = hostTemplate.resolve(context, httpServer, hostName);
-        executeAddAction(address);
-        return hostName;
-    }
-
-    public void removeHTTPServerHost(String httpServer, String ajpListener) throws InterruptedException, TimeoutException, IOException {
-        ResourceAddress address = hostTemplate.resolve(context, httpServer, ajpListener);
-        executeRemoveAction(address);
+    public String createWorker() throws InterruptedException, TimeoutException, IOException {
+        String name = "UndertowWorker_" + RandomStringUtils.randomAlphanumeric(6);
+        Address address = Address.subsystem("io").and("worker", name);
+        operations.add(address);
+        return name;
     }
 
-    public void removeHTTPServerHostIfExists(String httpServer, String ajpListener) throws InterruptedException, TimeoutException, IOException {
-        ResourceAddress address = hostTemplate.resolve(context, httpServer, ajpListener);
-        if (executeReadResourceAction(address)) {
-            executeRemoveAction(address);
+    /**
+     * If keepReference is set to true, name of created worker name will be kept internally in UndertowOperations
+     * @param keepReference whether the name of newly created worker will be saved
+     */
+    public String createWorker(boolean keepReference) throws InterruptedException, TimeoutException, IOException {
```

Comment:
Is `keepReference` argument necessary? Dou you really expect false value to be passed?